docs/design/object-locking.md

```diff
@@ -0,0 +1,187 @@
+# Object Locking using S3 Object Lock
+
+Object locking feature is built for addressing use cases where
+Write Once Read Many (WORM) model is required. The feature will be built in line
+with AWS specification, any extensions to the specification will be explicitly
+documented. The feature implementation has the goal of meeting
+SEC 17a-4 compliance.
+
+## Requirements
+
+* Bucket needs to have object lock flag set during bucket creation
+* Versioning has to be enabled on the bucket
+* Lock configuration can be written only on a bucket using PUT Object Lock
+  Configuration api that has object lock flag set
+
+PS: AWS specification does not have a way of setting object lock on existing
```

Comment:
The AWS spec....
Also would it be beneficial to mention that there is a way to set object lock on existing buckets using the `--token` flag, but that we're going to deviate by using a script?

Reply:
Agreed

Reply:
Added a section for deviation in spec.